### ISSUE TITLE
v7, HLS not work on player

### ISSUE THREAD

**Issue by shanka12** [pascalbaljetmedia/laravel-ffmpeg#191]
I'm using v7.

```
$lowBitrate  = (new X264 ('aac'))->setKiloBitrate(250)->setVideoCodec('libx264');
$midBitrate  = (new X264 ('aac'))->setKiloBitrate(500)->setVideoCodec('libx264');
$highBitrate = (new X264 ('aac'))->setKiloBitrate(1000)->setVideoCodec('libx264');

FFMpeg::fromDisk('videos_disk')
     ->open($this->video->original_file_path)
      ->exportForHLS()
      ->addFormat($lowBitrate, function($media) {
          $media->scale(480, 360);
       })
       ->addFormat($midBitrate, function($media) {
               $media->scale(640, 480);
       })
       ->addFormat($highBitrate, function($media) {
                $media->scale(1280, 720);
        })
       ->toDisk('streamable_videos')
                ->save($this->video->name . '/' . $this->video->video_id . '/' . $this->video->video_id . '.m3u8');

```
location of **streamable_videos** 
/storage/app/public/project1/stream

**problem is** 
**in v7, each format ts file, m3u8 file storing into _stream_ folder. final output file storing into the video id folder like below path**
```
->save($this->video->name . '/' . $this->video->video_id . '/' . $this->video->video_id . '.m3u8');
```
I'm using this final output m3u8 file on player. but video not playing. when I move all the ts files into this folder, it is working. I did not face this issue on previous version.

**Comment by pascalbaljet** [comment]
Could you run `composer update` and test again? I've made patch for this bug.

**Comment by shanka12** [comment]
> Could you run `composer update` and test again? I've made patch for this bug.
It is fixed now. Thank you :) 

I'm facing another issue. audio is not working.

**Comment by pascalbaljet** [comment]
So the export completes without an error or exception, the video works in the player but the audio doesn't? Which player do you use?

**Comment by shanka12** [comment]
no any error log. video works without souund.
clappr player
MacOS
Tested on safari, google chrome. tested on android browser also

**Comment by shanka12** [comment]
@pascalbaljet please let me know if you got solution for this audio issue

**Comment by swaraasolutions** [comment]
@shanka12 i faced the same issue while generating the HLS.

I modified this line and force added audio map

edit line 197 src/Exporters/HLSExporter.php
change from this 
`$this->addFormatOutputMapping($format, $disk->makeMedia($formatPlaylistPath), [$keysWithFilters[$key] ?? '0']);`
to this
`$this->addFormatOutputMapping($format, $disk->makeMedia($formatPlaylistPath), [$keysWithFilters[$key] ?? '0', "a:0"]);`

was able to get the audio back

@pascalbaljet i haven't gone through the entire source and not sure if this is right but have just patched it for timebeing.

**Comment by shanka12** [comment]
> @shanka12 i faced the same issue while generating the HLS.
> 
> I modified this line and force added audio map
> 
> edit line 197 src/Exporters/HLSExporter.php
> change from this
> `$this->addFormatOutputMapping($format, $disk->makeMedia($formatPlaylistPath), [$keysWithFilters[$key] ?? '0']);`
> to this
> `$this->addFormatOutputMapping($format, $disk->makeMedia($formatPlaylistPath), [$keysWithFilters[$key] ?? '0', "a:0"]);`
> 
> was able to get the audio back
> 
> @pascalbaljet i haven't gone through the entire source and not sure if this is right but have just patched it for timebeing.

@swaraasolutions Thank you :) I will check this.

**Comment by pascalbaljet** [comment]
@swaraasolutions thanks! Working on a patch :)

**Comment by pascalbaljet** [comment]
Waiting on the tests to complete and then I'll tag v7.0.2: https://github.com/pascalbaljetmedia/laravel-ffmpeg/pull/196

**Comment by shanka12** [comment]
I tested with @swaraasolutions.  Audio is working :)

**Comment by pascalbaljet** [comment]
Thanks for reporting, I've merged the patch and tagged v7.0.2.

**Comment by shanka12** [comment]
> Thanks for reporting, I've merged the patch and tagged v7.0.2.

Thank you @pascalbaljet :)
